### DISCUSSION TITLE
Suddenly stopped working

### DISCUSSION THREAD

**Discussion by JayBaywatch** [microsoft/azuredatastudio#19366] [Q&A]
Azure Data Studio on OSX suddenly stopped working for me today (2022/05/12) after a reboot of my macbook.
I tried installing the latest insiders build but get the same error, "Failed to start SQL Tools Service"".

Reviewing the console in devtools, I see the following which looks like it could be the issue.

common.css:1 Failed to load resource: net::ERR_FILE_NOT_FOUND
console.ts:137 [Extension Host] (node:3692) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.(Use `Azure Data Studio Helper (Renderer) --trace-deprecation ...` to show where the warning was created) (at writeOut (internal/process/warning.js:43:3))
g @ console.ts:137
console.ts:137 [Extension Host] Initialized Azure account extension storage. (at /private/var/folders/_4/bnnlv5qx61nc7b7z2w0_26_0l403lw/T/AppTranslocation/064FEB48-3A28-4E4F-862C-7BF51565AFAC/d/Azure Data Studio - Insiders.app/Contents/Resources/app/extensions/azurecore/dist/extension.js:2:674542)
log.ts:301   ERR An unknown error occurred. Please consult the log for more details.
log.ts:301   ERR Cannot read property 'length' of undefined: TypeError: Cannot read property 'length' of undefined
    at Object.getChildren (vscode-file://vscode-app/private/var/folders/_4/bnnlv5qx61nc7b7z2w0_26_0l403lw/T/AppTranslocation/064FEB48-3A28-4E4F-862C-7BF51565AFAC/d/Azure%20Data%20Studio%20-%20Insiders.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:2537:21492)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
notificationsAlerts.ts:40 **Failed to start SQL Tools Service**
onDidChangeNotification @ notificationsAlerts.ts:40
mainThreadExtensionService.ts:95 Activating extension 'Microsoft.mssql' failed: EROFS: read-only file system, mkdir '/private/var/folders/_4/bnnlv5qx61nc7b7z2w0_26_0l403lw/T/AppTranslocation/064FEB48-3A28-4E4F-862C-7BF51565AFAC/d/Azure Data Studio - Insiders.app/Contents/Resources/app/extensions/mssql/sqltoolsservice/Unknown/3.0.0-release.247'.
$onExtensionActivationError @ mainThreadExtensionService.ts:95

[vscode-app-1652403552141.log](https://github.com/microsoft/azuredatastudio/files/8683348/vscode-app-1652403552141.log)

Any ideas on what's going on here and maybe a push in the right direction?

**Comment by kburtram** [comment]
@JayBaywatch the problem appears to be the path to the SQL Tools Service is incorrect.  It says `'/private/var/folders/_4/bnnlv5qx61nc7b7z2w0_26_0l403lw/T/AppTranslocation/064FEB48-3A28-4E4F-862C-7BF51565AFAC/d/Azure Data Studio - Insiders.app/Contents/Resources/app/extensions/mssql/sqltoolsservice/Unknown/3.0.0-release.247`  but the `Unknown` folder should be the OS, so `macos` (or similar) in this case.  We detect the OS using a node module. I'll take a look at the details.

Can you provide more specific details on your version of mac OS?  Thanks!

**Reply by JayBaywatch** [reply]
Thanks for the response!

I'm running an 2020 MBP M1 - This has worked for months, until today I noticed there was a "preview" icon in the top left corner of the app, so I decided to check for updates via the ADS toolbar, but it reported there were no updates available. I rebooted an hour later and it didn't work, so I ended up uninstalling and installing the latest insiders build but that didn't solve my problem.

![image](https://user-images.githubusercontent.com/43705606/168192595-567b26c8-5533-446a-877f-1a04c2e01dad.png)

**Reply by JayBaywatch** [reply]
Looks like i have both and OSX and an Unknown folder at that location

![image](https://user-images.githubusercontent.com/43705606/168193246-fa49eb95-106e-4239-9e12-afff203bc682.png)

**Reply by kburtram** [reply]
Thanks for the additional info.  ADS must be creating that Unknown folder at runtime but unable to download the bits since we don't publish.  You may be able to copy everything from the OSX folder into Unknown to workaround but that doesn't seem like a good solution.

We use os.platform() to detect the OS, using this method https://github.com/microsoft/ads-service-downloader/blob/64463da626fe9b6c5ddb21361e3207ba0a6f94fb/src/platform.ts#L304.  I'm not sure why that would fail.  If I was guessing it could be a security issue with the macOS sandbox.  Have you tried moving the application into the Application directory to remove the mac quarantine bit (or otherwise remove this flag) and allow the app to run outside the sandbox?

**Reply by JayBaywatch** [reply]
Also, as a temporary workaround, I renamed the 3.0.0-release.238 directory under Unknown to 3.0.0-release.238.bak and then copied the 3.0.0-release.238 sub-directory from the OSX folder and I am able to connect again. The PG service still fails but I don't have any pressing Postgres work due for a few weeks, so I can live with that for now.

**Reply by JayBaywatch** [reply]
Thanks, I did receive a Zcaler update today. I'll check with OpSec and inquire if they pushed any security updates that could be triggering this. I appreciate you. Thanks!

**Reply by JayBaywatch** [reply]
One more piece of info. I installed the latest version of node for apple silcon just now and I do get the correct platform name back.

![image](https://user-images.githubusercontent.com/43705606/168196505-cc15938b-4dd6-4037-a05d-f03078035f45.png)

I am assuming that ADS bundles its own version of node, is there any chance it got mangled for me somehow?

![image](https://user-images.githubusercontent.com/43705606/168196707-33048f6d-cf7c-4d7e-92b7-883f16e66772.png)

Sorry, I'm not very versed in node, just starting with the obvious I guess.

**Reply by kburtram** [reply]
Thanks for the additional information.  that `os.platform` call is basically what we do in ADS, typically the version of the module shouldn't matter.  When the app has a file path starting with something like `/private/var/folders/_4/bnnlv5qx61nc7b7z2w0_26_0l403lw/T/AppTranslocation` it's being sandboxed by the macos gatekeeper security feature, which shouldn't matter, except autoupdate typically doesn't work.  My guess was that either that or some other security mechanism is causing that call to fail in that particular security context (running node from CLI shouldn't get sandboxed so would be a different execution context).  In any case, that is just a guess since I don't have a local repro.  I'll think about how to confirm & mitigate.
